docs/en/Unit-Testing.md

```diff
@@ -0,0 +1,78 @@
+---
+layout: global
+title: Unit Testing Guidelines
+nickname: Unit Testing Guidelines
+group: Resources
+---
+
+* Table of Contents
+{:toc}
+
+# Unit testing goals
+1. Unit tests act as examples of how to use the code under test.
+2. Unit tests detect when an object breaks it's specification.
+3. Unit tests *don't* break when an object is refactored but still meets the same specification.
+
+# How to write a unit test
+
+1. If creating an instance of the class takes some work, create `@Before` and `@After` methods which handle setup and cleanup. These methods get run automatically before and after each unit test. Only do general setup which will apply to every test. Test-specific setup should be done locally in the tests that need it. In this example, we are testing a `BlockMaster`, which depends on a journal, clock, and executor service. The executor service and journal we provide are real implementations, and the `TestClock` is a fake clock which can be controlled by unit tests.
```

Comment:
I wouldn't encourage `@BeforeClass`/`@AfterClass` since it makes it easier for tests to interfere with each other, but I also wouldn't want to ban it outright since it can be useful when used right.